lib/block_view_flush.go

```diff
@@ -1021,46 +997,24 @@ func (bav *UtxoView) _flushMessagingGroupEntriesToDbWithTxn(txn *badger.Txn, blo
 			//
 			// TODO: We should have a single PutMappings function in db_utils.go that we push this
 			// complexity into.
-			ownerPublicKey := &messagingGroupKey.OwnerPublicKey
+			ownerPublicKey := messagingGroupKey.OwnerPublicKey
 			if err := DBPutMessagingGroupEntryWithTxn(txn, bav.Snapshot, blockHeight,
-				ownerPublicKey, messagingGroupEntry); err != nil {
+				&ownerPublicKey, messagingGroupEntry); err != nil {
 				return errors.Wrapf(err, "UtxoView._flushMessagingGroupEntriesToDbWithTxn: "+
-					"Fail while putting group entry. Problem putting MessagingGroupEntry %v to db", *messagingGroupEntry)
+					"Problem putting MessagingGroupEntry %v to db", *messagingGroupEntry)
 			}
-			if blockHeight >= uint64(bav.Params.ForkHeights.DeSoUnlimitedDerivedKeysAndMessagesMutingAndMembershipIndexBlockHeight) {
-				// Group owner can be one of the group members, particularly when we want to add the
+			for _, recipient := range messagingGroupEntry.MessagingGroupMembers {
+				// Group owner can be one of the recipients, particularly when we want to add the
 				// encrypted key addressed to the owner. This could happen when the group is created
 				// by a derived key, and we want to allow the main owner key to be able to read the chat.
-				// After the muting and prefix optimization block height, we will store the owner's groups
-				// in the membership index by default. This way we can easily query for the messaging group
-				// simplified entry in the db, saving read time, e.g. when we want to send a message to the group.
-
-				if err := DBPutMessagingGroupOwnerInMembershipIndexWithTxn(txn, bav.Snapshot, blockHeight,
-					ownerPublicKey, messagingGroupEntry); err != nil {
-					return errors.Wrapf(err, "UtxoView._flushMessagingGroupEntriesToDbWithTxn: "+
-						"Fail while putting owner membership index. Problem putting MessagingGroupEntry %v to db", *messagingGroupEntry)
+				if reflect.DeepEqual(recipient.GroupMemberPublicKey[:], ownerPublicKey[:]) {
+					continue
```

Comment:
this looks like we exclude the group owner from the db, is this right?

Reply:
yes, I reverted to what we had in main before the membership index.